src/me/remie/vulcan/leaguetasks/utils/LeaguePanel.java

```diff
@@ -0,0 +1,119 @@
+package me.remie.vulcan.leaguetasks.utils;
+
+import simple.hooks.simplebot.Game;
+import simple.hooks.wrappers.SimpleWidget;
+import simple.robot.api.ClientContext;
+
+public class LeaguePanel {
+    public ClientContext ctx;
+
+    public LeaguePanel(final ClientContext ctx) {
+        this.ctx = ctx;
+    }
+
+    public boolean doWeNeedToClearFilters() {
+        final SimpleWidget tierDropDownMenu = ctx.widgets.getWidget(657,27);
+        if (!isOptionSelected(tierDropDownMenu, 4, "all")) return true;
+
+        final SimpleWidget typeDropDownMenu = ctx.widgets.getWidget(657,28);
+        if (!isOptionSelected(typeDropDownMenu, 4, "all")) return true;
+
+        final SimpleWidget areaDropDownMenu = ctx.widgets.getWidget(657,29);
+        if (!isOptionSelected(areaDropDownMenu, 4, "all")) return true;
+
+        final SimpleWidget completedDropDownMenu = ctx.widgets.getWidget(657,31);
+        if (!isOptionSelected(completedDropDownMenu, 4, "all")) return true;
+        return false;
+    }
+
+    private boolean isOptionSelected(SimpleWidget widget, int childIndex, String option) {
+        if (widget == null || widget.getChild(childIndex) == null) return false;
+        return widget.getChild(childIndex).getText().equalsIgnoreCase(option);
+    }
+
+    public boolean clearAllFilterOptions() {
+        if (!isLeaguePanelOpen()) return false;
```

Comment:
```suggestion
```
I'm wondering if this is even needed, as we're only calling this function right after opening/checking if the panel is open already.

Reply:
It was for sanity check. Np removing it.